app/services/room_settings_getter.rb

```diff
@@ -23,6 +24,12 @@ def initialize(room_id:, provider:, settings: [], show_codes: false, only_enable
   end
 
   def call
+    # checking if CanRecord permission is set to true when RoomConfig record is optional
+    if @rooms_configs['record'].nil? && (RolePermission.joins(:permission).find_by(role_id: @current_user.role_id,
+                                                                                   permission: { name: 'CanRecord' })&.value == 'false')
+      @rooms_configs['record'] = 'false'
+    end
```

Comment:
I would move this logic down below the call to `infer_codes` just so that it doesn't get overwritten. I would also move it to a private method to make it a bit cleaner to read. 

```
def infer_delete(room_settings:)
  return if RolePermission.joins(:permission).find_by(role_id: @current_user.role_id, permission: { name: 'CanRecord' }) == 'true'

  room_settings['record'] = false
end
```

Reply:
Something like this (needs to be tested) ^

Reply:
Yup, tests will be part of next PR